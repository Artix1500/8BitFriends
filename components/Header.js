import React from 'react';
import {View, Text, StyleSheet, Button} from 'react-native';

const Header = ({title, logout, navigation}) => {
  return (
    <View style={styles.header}>
      <Text style={styles.text}>{title}</Text>
      {logout ? (
        <Button
          style={styles.logout}
          color="white"
          title="Logout"
          onPress={() => navigation.navigate('Login')}
        />
      ) : (
        <React.Fragment />
      )}
    </View>
  );
};

Header.defaultProps = {
  title: '8BitFriends',
  logout: false,
  navigation: null,
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    height: 65,
    padding: 15,
    backgroundColor: 'indigo',
  },
  text: {
    color: 'white',
    fontSize: 23,
    textAlign: 'center',
  },
  logout: {
    color: 'white',
    fontSize: 12,
    textAlign: 'left',
  },
});

export default Header;
