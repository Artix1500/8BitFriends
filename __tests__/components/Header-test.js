/**
 * @format
 */

import 'react-native';
import React from 'react';
import Header from '../../components/Header';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Header Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders title from props', () => {
    const tree = renderer.create(<Header title={'Example'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
