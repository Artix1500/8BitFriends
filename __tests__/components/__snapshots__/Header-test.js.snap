// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "indigo",
      "height": 60,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 23,
        "textAlign": "center",
      }
    }
  >
    8BitFriends
  </Text>
</View>
`;

exports[`Header Component renders title from props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "indigo",
      "height": 60,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 23,
        "textAlign": "center",
      }
    }
  >
    Example
  </Text>
</View>
`;
